test/files/neg/t4940.scala

```diff
@@ -0,0 +1,18 @@
+// scalac: -Werror -Xlint
+class C {
```

Comment:
Scala 3 gives

```
-- [E172] Type Error: ../test/files/neg/t4940.scala:3:67 -----------------------
3 |  val f: PartialFunction[String, Int] = (x: Int) => x match { case "x" => 3 }   // error
  |                                                                   ^^^
  |           Values of types String and Int cannot be compared with == or !=
-- [E007] Type Mismatch Error: ../test/files/neg/t4940.scala:5:65 --------------
5 |  val g: PartialFunction[String, Int] = (x: X) => x match { case "x" => 3 }     // error
  |                                                                 ^^^
  |                           Found:    ("x" : String)
  |                           Required: X
  |                           pattern type is incompatible with expected type
  |
  | longer explanation available when compiling with `-explain`
-- [E007] Type Mismatch Error: ../test/files/neg/t4940.scala:7:37 --------------
7 |  val m: PartialFunction[Int, Int] = (x: Double) => x match { case 3.14 => 3 }  // error
  |                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |                                     Found:    Double => Int
  |                                     Required: PartialFunction[Int, Int]
  |
  | longer explanation available when compiling with `-explain`
3 errors found
```

Reply:
while I think of it, thanks for taking a look at draft PRs, in case they have a future. It's a bit like you're in my bathroom pointing out that there is hair in the drain, except in code it's like, I don't remember how a drain works, so let me revisit it.

I remember that scala 3 is very simple for PFs so I wondered why isn't S2 so simple and uniform.

Reply:
Seth and me had a quick chat about draft PRs, we don't mind keeping them around and open. But we'll generally skip over them when working on the PR queue. I still might peek like I did here.